gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineExactDependencies.java

```diff
@@ -119,80 +116,95 @@ private static void configureSourceSet(
+                            conf.extendsFrom(project.getConfigurations().getByName("lockConstraints"));
+                        }
+                        // Inherit the excludes from compileClasspath too (that get aggregated from all its
+                        // super-configurations).
+                        compileClasspath.get().getExcludeRules().forEach(rule -> conf.exclude(excludeRuleAsMap(rule)));
+                    });
 
-        // Since we are copying configurations before resolving 'explicitCompile', make double sure that it's not
-        // being resolved (or dependencies realized via `.getIncoming().getDependencies()`) too early.
-        AtomicBoolean projectsEvaluated = new AtomicBoolean();
-        project.getGradle().projectsEvaluated(g -> projectsEvaluated.set(true));
-        explicitCompile
-                .getIncoming()
-                .beforeResolve(ir -> Preconditions.checkState(
-                        projectsEvaluated.get()
-                                || (project.getGradle().getStartParameter().isConfigureOnDemand()
-                                        && project.getState().getExecuted()),
-                        "Tried to resolve %s too early.",
-                        explicitCompile));
+                    // Since we are copying configurations before resolving 'explicitCompile', make double sure that
+                    // it's not
+                    // being resolved (or dependencies realized via `.getIncoming().getDependencies()`) too early.
+                    AtomicBoolean projectsEvaluated = new AtomicBoolean();
+                    project.getGradle().projectsEvaluated(g -> projectsEvaluated.set(true));
+
+                    conf.getIncoming().beforeResolve(_ignored -> {
+                        Preconditions.checkState(
+                                projectsEvaluated.get()
+                                        || (project.getGradle()
+                                                        .getStartParameter()
+                                                        .isConfigureOnDemand()
+                                                && project.getState().getExecuted()),
+                                "Tried to resolve %s too early.",
+                                conf);
+                    });
+                });
 
         TaskProvider<CheckUnusedDependenciesTask> sourceSetUnusedDependencies = project.getTasks()
                 .register(
                         checkUnusedDependenciesNameForSourceSet(sourceSet), CheckUnusedDependenciesTask.class, task -> {
                             task.dependsOn(sourceSet.getClassesTaskName());
                             task.setSourceClasses(sourceSet.getOutput().getClassesDirs());
-                            task.dependenciesConfiguration(explicitCompile);
+                            task.getDependenciesConfigurations().add(explicitCompile);
```

Comment:
wacky idea, not actually sure, but can't we just use the `getHierarchy()` method of the configuration we're initially copying? and then take the first one? do we actually need/want a copy, or did we only take a copy because that was the only way to get the first level without the other dependencies from super configurations?

Reply:
because with that, you can just do a `conf.map(c -> Iterables.getFirst(c.getHierarchy()))`, it's also a `LinkedSet` populated in DFS manner so in theory it should have what we want assuming we don't want/need a separate copy

Reply:
Not really sure I follow here, the first time I saw this plugin was this PR. I'd like to focus on getting this to work at all with Gradle 8 first, if we want to improve it/change it in some way it seems less risky to spend time on that intentionally in another change.